server/src/main/java/org/elasticsearch/indices/recovery/MultiFileWriter.java

```diff
@@ -210,7 +216,9 @@ void writeChunk(FileChunk newChunk) throws IOException {
                     }
                     pendingChunks.remove();
                 }
-                innerWriteFileChunk(chunk.md, chunk.position, chunk.content, chunk.lastChunk);
+                try (chunk) {
+                    innerWriteFileChunk(chunk.md, chunk.position, chunk.content, chunk.lastChunk);
+                }
                 synchronized (this) {
                     assert lastPosition == chunk.position : "last_position " + lastPosition + " != chunk_position " + chunk.position;
                     lastPosition += chunk.content.length();
```

Comment:
The chunk is already closed here, seems wrong to be accessing it still. Can we close it after the `synchronised` block?

Reply:
++ that was a little dirty, I made it close later now as suggested